### ISSUE TITLE
Scoping issues with default parameters

### ISSUE THREAD

**Issue by gabejohnson** [sweet-js/sweet.js#678]
```js
function foo(a = 1, b = a) { return a + b; }
```
should result in:
```js
function foo(a_100 = 1, b_101 = a_100) {
  return a_100 + b_101;
}
```
but instead results in:
```js
function foo(a_100 = 1, b_101 = a) { // note that 'a' has not been gensym'd
  return a_100 + b_101;
}
```

@disnet, is this an issue in `TermExpander`?

**Comment by disnet** [comment]
Yeah right around [here](https://github.com/sweet-js/sweet.js/blob/bcbfa484bc7ee92f975aa050c5b205971e4c2e2c/src/term-expander.js#L550-L580) it looks like the hygiene stuff isn't taking into account initializers.

It needs to apply the function scope in initializers.

Also, I think the current CloneReducer is potentially picking up too many bindings so that should be fixed too.

**Comment by gabejohnson** [comment]
There's also a problem as those params aren't getting parsed as `BindingWithDefault`s.
